src/Entity/ItemId.php

```diff
@@ -95,16 +96,18 @@ public function unserialize( $serialized ) {
 	 * serialization. Not doing so in new code requires special justification.
 	 *
 	 * @param int|float|string $numericId
+	 * @param string $repositoryName, defaults to an empty string (local repository)
 	 *
 	 * @return self
 	 * @throws InvalidArgumentException
 	 */
-	public static function newFromNumber( $numericId ) {
+	public static function newFromNumber( $numericId, $repositoryName = '' ) {
 		if ( !is_numeric( $numericId ) ) {
 			throw new InvalidArgumentException( '$numericId must be numeric' );
 		}
+		RepositoryNameAssert::assertParameterIsValidRepositoryName( $repositoryName, '$repositoryName' );
```

Comment:
In foo:bar:Q123, the repo name is "foo", not "foo:bar". The relative ID, the one that can be resolved by repo foo, is "bar:Q123".

By our spec, repo names cannot contain a colon, and EntityId::getRepositoryName() will never retrun a string that contains a column.